components/raftstore/src/store/peer.rs

```diff
@@ -2427,10 +2427,9 @@ where
             );
             poll_ctx.raft_metrics.invalid_proposal.read_index_no_leader += 1;
             // The leader may be hibernated, send a message for trying to awaken the leader.
-            if poll_ctx.cfg.hibernate_regions
-                && (self.bcast_wake_up_time.is_none()
-                    || self.bcast_wake_up_time.as_ref().unwrap().elapsed()
-                        >= Duration::from_millis(MIN_BCAST_WAKE_UP_INTERVAL))
+            if self.bcast_wake_up_time.is_none()
```

Comment:
If current peer is not leader, it's not reasonable to response error `can not read index due to no leader`. I think it's better to wait until find its leader and continue process.

Reply:
> response region error (just like NotLeader with empty data)

I take this approach.